submissions/pozzitive11/js-world-oop/index.js

```diff
@@ -0,0 +1,65 @@
+class Inhabitant {
+  constructor(species, name, gender, saying) {
+    this.species = species;
+    this.name = name;
+    this.gender = gender;
+    this.saying = saying;
+  }
+
+  getProperties() {
+    return `${this.species}; My name ${this.name}; ${this.gender}; ${this.saying}`;
+  }
+}
+class Animal extends Inhabitant {
+  constructor(name, gender, saying) {
+    super("animal", name, gender, saying);
+    this.paw = 4;
+  }
+  getProperties() {
+    return super.getProperties() + `; I have ${this.paw} paws;`;
+  }
+}
+
+class Human extends Inhabitant {
+  constructor(name, gender, saying) {
+    super("human", name, gender, saying);
+    this.hands = 2;
+    this.legs = 2;
+  }
+
+  getProperties() {
+    return (
+      super.getProperties() +
+      `; I have ${this.legs} legs and ${this.hands} hands;`
+    );
+  }
+}
+
+class Dog extends Animal {
+  constructor(name, gender) {
+    super(name, gender, "Woof-woof!");
+  }
+}
+
+class Cat extends Animal {
+  constructor(name, gender) {
+    super(name, gender, "meow-meow");
+  }
+}
```

Comment:
The thing is that while `man` and `woman` are not species but rather genders within the same species, dogs and cats are distinct species. Need to fix.